commands/fork.go

```diff
@@ -112,16 +112,15 @@ func fork(cmd *Command, args *Args) {
 		originURL := originRemote.URL.String()
 		url := forkProject.GitURL("", "", true)
 
-		// Check to see if the remote already exists and points to the fork.
-		currentRemote, err := localRepo.RemoteByName(newRemoteName)
+		_, err := localRepo.RemoteByName(newRemoteName)
 		if err == nil {
-			currentProject, err := currentRemote.Project()
-			if err == nil {
-				if currentProject.SameAs(forkProject) {
-					ui.Printf("existing remote: %s\n", newRemoteName)
-					return
-				}
+			if newRemoteName != "origin" {
+				ui.Printf("existing remote: %s\n", newRemoteName)
+				return
 			}
+			// Assume user wants to follow github guides for collaboration
+			ui.Printf("renaming %s as upstream\n", newRemoteName)
+			args.Before("git", "remote", "rename", newRemoteName, "upstream")
```

Comment:
What if `upstream` already exists?

Reply:
hub will fail because of git: upstream already exists. Actually Printf of origin being renamed is there to allow user understand that potential git error.